src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1324,6 +1325,72 @@ def create_or_replace_view(
             source_plan,
         )
 
+    def find_table_function_in_sql_tree(
+        self, plan: SnowflakePlan
+    ) -> Optional[SnowflakePlan]:
+        """This function is meant to find any udtf function call from a create dynamic table plan and
+        replace '*' with explicit column identifier in the select of table function. Since we cannot
+        differentiate udtf call from other table functions, we apply this change to all table functions.
+        """
+        deepcopied_plan = copy.deepcopy(plan)
+        queue = deque()
+        queue.append(deepcopied_plan)
+        from snowflake.snowpark._internal.analyzer.select_statement import (
+            SelectTableFunction,
+            Selectable,
+        )
+
+        while queue:
+            plan_of_child = queue.popleft()
+            for node in plan_of_child.children_plan_nodes:
+                queue.append(node)
+
+            # the bug only happen when create dynamic table on top of a table function
+            # this is meant to decide whether the plan is select from a table function
+            if isinstance(plan_of_child, SelectTableFunction) and isinstance(
+                plan_of_child.snowflake_plan.source_plan, TableFunctionJoin
+            ):
+                # if clause used to decide that right column is '*' that we need to change and there is only 1 child
+                # plan to change, a table function can only be right joined, so we only care about right column here.
+                if (
+                    plan_of_child.snowflake_plan.source_plan.right_cols == ["*"]
+                    and len(plan_of_child.snowflake_plan.children_plan_nodes) == 1
+                ):
+                    child_plan = plan_of_child.snowflake_plan.children_plan_nodes[0]
+                    if isinstance(child_plan, Selectable):
+                        child_plan = child_plan.snowflake_plan
```

Comment:
```suggestion
                    if isinstance(child_plan, Selectable):
                        child_plan = child_plan.snowflake_plan
                    if not isinstrance(child, SnowflakePlan):
                        return None
```
here the assumption is that child_plan is `SnowflakePlan` object. We should either return or `assert isinstance(child_plan, SnowflakePlan)`